modules/reporting/src/main/java/com/devonfw/module/reporting/common/config/ReportExporterConfiguration.java

```diff
@@ -0,0 +1,217 @@
+package com.devonfw.module.reporting.common.config;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import net.sf.jasperreports.engine.JRAbstractExporter;
+import net.sf.jasperreports.engine.export.HtmlExporter;
+import net.sf.jasperreports.engine.export.JRCsvExporter;
+import net.sf.jasperreports.engine.export.JRPdfExporter;
+import net.sf.jasperreports.engine.export.JRRtfExporter;
+import net.sf.jasperreports.engine.export.JRTextExporter;
+import net.sf.jasperreports.engine.export.JRXlsExporter;
+import net.sf.jasperreports.engine.export.oasis.JROdsExporter;
+import net.sf.jasperreports.engine.export.oasis.JROdtExporter;
+import net.sf.jasperreports.engine.export.ooxml.JRDocxExporter;
+import net.sf.jasperreports.engine.export.ooxml.JRPptxExporter;
+import net.sf.jasperreports.engine.export.ooxml.JRXlsxExporter;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Configuration class that returns the {@link JRAbstractExporter exporter} according to the {@link ReportingConstants}
+ *
+ */
+@Configuration
+public class ReportExporterConfiguration {
+
+  private final Map<String, JRAbstractExporter> exporterMap;
+
+  /**
+   * The constructor.
+   */
+  public ReportExporterConfiguration() {
+
+    super();
+    this.exporterMap = new HashMap<>();
+  }
+
+  /**
+   * puts the {@link JRAbstractExporter} Instance in to the {@link Map} according to {@link ReportingConstants}
+   *
+   * @param jrAbstractExporters
+   */
+  @Inject
+  public void setReportExporters(List<JRAbstractExporter> jrAbstractExporters) {
+
+    for (JRAbstractExporter exporter : jrAbstractExporters) {
+      this.exporterMap.put(exporter.getExporterKey(), exporter);
+    }
+
+  }
+
+  @Bean
+  public JRAbstractExporter getExcelExporter() {
+
+    return new JRXlsExporter() {
+      @Override
+      public String getExporterKey() {
+
+        return ReportingConstants.XLS;
+      }
+    };
+  }
+
+  /**
+   * Returns the {@link JRAbstractExporter} for Xls version
+   *
+   * @return
+   */
+  @Bean
+  public JRAbstractExporter getPdfExporter() {
+
+    return new JRPdfExporter() {
```

Comment:
...
This looks like we turned a plugable concept of jasper into a monolithic approach. Does not seem to be smart to me. Am I missing something...?!? And if Jasper is providing such exporters as plugins like `JRXlsExporter` and `JRPdfExporter`, etc. why on earth do we have to override `getExporterKey()` method from outside. This appears strange to me. However, I am reviewing Jasper now while I actually should review this devon4j module instead...

Reply:
Also interesting that for `getPdfExporter` we have JavaDoc while for the others we dont. Further again this is the JavaDoc antipattern of
```
* Returns ...
* @return 
```
So simply write what the method returns after the `@return` tag withour the `Returns`. This is how JavaDoc works. I know that you already know and just took over the code from `devon` repo. However, I am seeing this a lot in reviews. Maybe we should create a better DoD or do some trainings for all the involved coders to avoid talking about basics in reviews.